### ISSUE TITLE
combo-box updating it's items but not it's selected item

### ISSUE THREAD

**Issue by torkus** [cljfx/cljfx#102]
I have a combo-box that will update it's items but not it's label, resulting in a combo-box with a blank label.

It's easy to recreate and I have an example here: 

```
git clone https://github.com/ogri-la/cljfx-issues
cd cljfx-issues
lein run
```

In the example there are three combo-boxes each configured slightly differently but all failing to render their label when the underlying list of items changes.

**Comment by vlaaad** [comment]
Hmm, I thought specifying `:button-cell` and `:cell-factory` should be enough for combo-boxes, but apparently it's not — I reproduced the issue. It seems that the combo-box value gets silently reset to `nil` when `:items` change.

A workaround is `ext-recreate-on-key-changed` lifecycle described [here](https://github.com/cljfx/cljfx/issues/76#issuecomment-645563116), where `:key` should be the same value as in `:items`.

Example using this lifecycle:
```clj
(def state
  (atom {:values {:one "one"
                  :two "two"}
         :selected :one}))

(fx/mount-renderer
  state
  (fx/create-renderer
    :middleware
    (fx/wrap-map-desc
      (fn [{:keys [values selected] :as in}]
        {:fx/type :stage
         :showing true
         :width 600
         :height 500
         :scene
         {:fx/type :scene
          :root
          {:fx/type :v-box
           :children
           [{:fx/type ext-recreate-on-key-changed
             :key (sort-by key values)
             :desc {:fx/type :combo-box
                    :button-cell (fn [e] {:text (val e)})
                    :cell-factory {:fx/cell-type :list-cell
                                   :describe (fn [e] {:text (val e)})}
                    :value (find values selected)
                    :on-value-changed (fn [e] (swap! state assoc :selected (key e)))
                    :items (sort-by key values)}}
            {:fx/type :button
             :text "Add"
             :on-action (fn [_] (swap! state update :values assoc :three "three"))}
            {:fx/type :button
             :text "Remove"
             :on-action (fn [_] (swap! state update :values dissoc :three))}
            {:fx/type :label
             :text (with-out-str (clojure.pprint/pprint in))}]}}}))))
```

**Comment by torkus** [comment]
thanks for this @vlaaad , it fixes the problem I was having. I was also able to remove `:cell-factory` and `:button-cell` and have it work simply again across linux and mac.

**Comment by vlaaad** [comment]
Fixed for good in 1.7.12
